src/app/Notifications/NotificationCenter.tsx

```diff
@@ -37,35 +37,87 @@
  */
 import * as React from 'react';
 import { Dropdown, DropdownItem, DropdownPosition, KebabToggle,
-  NotificationDrawer, NotificationDrawerBody, NotificationDrawerHeader,
+  NotificationDrawer, NotificationDrawerBody, NotificationDrawerGroup, NotificationDrawerGroupList, NotificationDrawerHeader,
   NotificationDrawerList, NotificationDrawerListItem,
   NotificationDrawerListItemBody, NotificationDrawerListItemHeader, Text,
   TextVariants } from '@patternfly/react-core';
 import { Notification, NotificationsContext } from './Notifications';
+import { combineLatest } from 'rxjs';
 
 export interface NotificationCenterProps {
   onClose: () => void;
 }
 
+export interface NotificationDrawerCategory {
+  title: string;
+  isExpanded: boolean;
+  notifications: Notification[];
+  unreadCount: number;
+}
+
 export const NotificationCenter: React.FunctionComponent<NotificationCenterProps> = props => {
   const context = React.useContext(NotificationsContext);
-  const [notifications, setNotifications] = React.useState([] as Notification[]);
-  const [unreadNotificationsCount, setUnreadNotificationsCount] = React.useState(0);
+  const [totalUnreadNotificationsCount, setTotalUnreadNotificationsCount] = React.useState(0);
   const [isHeaderDropdownOpen, setHeaderDropdownOpen] = React.useState(false);
+  const PROBLEMS_CATEGORY_IDX = 2;
+  const [drawerCategories, setDrawerCategories] = React.useState([
+    {title: "Completed Actions", isExpanded: true, notifications: [] as Notification[], unreadCount: 0},
+    {title: "Network Info", isExpanded: false, notifications: [] as Notification[], unreadCount: 0},
+    {title: "Problems", isExpanded: false, notifications: [] as Notification[], unreadCount: 0}
+  ] as NotificationDrawerCategory[]);
+
+  const countUnreadNotifications = (notifications: Notification[]) => {
+    return notifications.filter(n => !n.read).length;
+  }
 
   React.useEffect(() => {
-    const sub = context.notifications().subscribe(setNotifications);
+    const sub = combineLatest([context.actionsNotifications(), context.networkInfoNotifications(), context.problemsNotifications()])
+    .subscribe(notificationLists => {
+        setDrawerCategories(drawerCategories => {
+
+          return drawerCategories.map((category: NotificationDrawerCategory, idx) => {
+            category.notifications = notificationLists[idx];
+            category.unreadCount = countUnreadNotifications(notificationLists[idx]);
+            return category;
+        }) as unknown as NotificationDrawerCategory[];
```

Comment:
What's up with this `as unknown as NotificationDrawerCategory[]`? That looks awkward.

Reply:
Whoops, that was leftover from when I was trying to fix a return type error. It turns out I was missing the `return category` line. I'll remove this.